### PR TITLE
[libzip] bump to 1.4.0

### PR DESCRIPTION
**Pull request by tobiaskohlbau**
Signed-off-by: Tobias Kohlbau <tobias@kohlbau.de>

### CODE REVIEW

**Comment by janisozaur** [comment]
@tobiaskohlbau I see you have made some changes that related to UWP builds, there were some fixes merged upstream to enable them here: https://github.com/nih-at/libzip/pull/16

Can you confirm all the UWP targets build properly?

File: ports/libzip/enable-static.patch [hunk 1]
```diff
@@ -15,30 +15,30 @@ index deceb65..1283013 100644
...
+index 0624a2b..06c8ecb 100644
 --- a/cmake-config.h.in
 +++ b/cmake-config.h.in
-@@ -60,7 +60,7 @@
- #cmakedefine WORDS_BIGENDIAN
+@@ -63,7 +63,7 @@
+ #cmakedefine HAVE_SHARED
  /* END DEFINES */
  #define PACKAGE "@PACKAGE@"
 -#define VERSION "@VERSION@"
```
Comment (janisozaur) [inline ports/libzip/enable-static.patch]: I don't think this part is needed anymore, it was addressed in https://github.com/nih-at/libzip/commit/85931c95f56f5a39b6f891bd0f35091fa525e616

File: ports/libzip/portfile.cmake [hunk 2]
```diff
@@ -1,9 +1,9 @@
 include(vcpkg_common_functions)
-set(SOURCE_PATH ${CURRENT_BUILDTREES_DIR}/src/libzip-1.3.2)
+set(SOURCE_PATH ${CURRENT_BUILDTREES_DIR}/src/libzip-1.4.0)
 vcpkg_download_distfile(ARCHIVE_FILE
```
Comment (janisozaur) [inline ports/libzip/portfile.cmake]: The project has now officially moved to github, there is a vcpkg call to fetch archives from here, can you update that as well?

File: ports/libzip/enable-static.patch [hunk 3]
```diff
@@ -47,10 +47,10 @@ index 0d1839c..dd450ff 100644
...
  #  if SIZE_T_LIBZIP == INT_LIBZIP
 diff --git a/cmake-zipconf.h.in b/cmake-zipconf.h.in
-index 6a276f6..410c898 100644
+index f0e24e8..e087bfc 100644
 --- a/cmake-zipconf.h.in
 +++ b/cmake-zipconf.h.in
-@@ -118,4 +118,10 @@ typedef unsigned long long zip_uint64_t;
+@@ -125,4 +125,10 @@ typedef unsigned long long zip_uint64_t;
  #define ZIP_INT64_MAX	 0x7fffffffffffffffLL
  #define ZIP_UINT64_MAX	 0xffffffffffffffffULL
```
Comment (janisozaur) [inline ports/libzip/enable-static.patch]: The part below should not be needed anymore, the CMake's `BUILD_SHARED_LIBS` should suffice, as addressed here: https://github.com/nih-at/libzip/commit/6b1e8de

File: ports/libzip/enable-static.patch [hunk 4]
```diff
@@ -62,7 +62,7 @@ index 6a276f6..410c898 100644
 +
  #endif /* zipconf.h */
 diff --git a/lib/compat.h b/lib/compat.h
-index 625c84e..8943587 100644
+index f2a5467..667f3dc 100644
 --- a/lib/compat.h
 +++ b/lib/compat.h
 @@ -42,7 +42,9 @@
```
Comment (janisozaur) [inline ports/libzip/enable-static.patch]: Likewise, this part below should not be needed anymore with fix in https://github.com/nih-at/libzip/commit/6b1e8de

File: ports/libzip/enable-static.patch [hunk 5]
```diff
@@ -1,8 +1,8 @@
 diff --git a/CMakeLists.txt b/CMakeLists.txt
-index deceb65..1283013 100644
+index 69cdb8b..8b3fbdb 100644
 --- a/CMakeLists.txt
 +++ b/CMakeLists.txt
-@@ -113,6 +113,15 @@ ADD_DEFINITIONS("-D_CRT_SECURE_NO_WARNINGS")
+@@ -133,6 +133,15 @@ ADD_DEFINITIONS("-D_CRT_SECURE_NO_WARNINGS")
  ADD_DEFINITIONS("-D_CRT_NONSTDC_NO_DEPRECATE")
  ENDIF(MSVC)
```
Comment (janisozaur) [inline ports/libzip/enable-static.patch]: The part below should be changed to simply set `BUILD_SHARED_LIBS` on or off, to accommodate fix in https://github.com/nih-at/libzip/commit/6b1e8de

**Comment by janisozaur** [comment]
I've left a few comments for you to address.

As the patch file should not be concerned with much about static builds, other than enabling them, I'd suggest renaming it as well.

zlib seems to have a [similar patch](https://github.com/Microsoft/vcpkg/blob/master/ports/zlib/cmake_dont_build_more_than_needed.patch) so I would say `cmake_dont_build_more_than_needed.patch` would be a good name.

**Comment by tobiaskohlbau** [comment]
I think you've mistaken the changelog within your first comment, I've not changed UWP settings it's only diff cause the patch file appends changes after that lines. I will address your comments and will update this PR.

File: ports/libzip/portfile.cmake [hunk 6]
```diff
@@ -1,34 +1,24 @@
 include(vcpkg_common_functions)
-set(SOURCE_PATH ${CURRENT_BUILDTREES_DIR}/src/libzip-1.3.2)
-vcpkg_download_distfile(ARCHIVE_FILE
-    URLS "https://nih.at/libzip/libzip-1.3.2.tar.gz"
-    FILENAME "libzip-1.3.2.tar.gz"
-    SHA512 75b7e6f541be30e721275723f264c20f9a3be5335d954b5909acdddb0f6dd9b2420166904c9b88206692a57a4aa54e4fe8ed4d62c1f4b900aebf6ad40f767376
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO nih-at/libzip
+    REF rel-1-4-0
```
Comment (janisozaur) [inline ports/libzip/portfile.cmake]: `PREFER_NINJA`

Comment (tobiaskohlbau) [reply ports/libzip/portfile.cmake]: Wrong place but I get it.

Comment (janisozaur) [reply ports/libzip/portfile.cmake]: 👍

**Comment by tobiaskohlbau** [comment]
I’ve found an issue which wasn’t noticed before as libzip calls `find_package(bzip2)` which was installed on my machine. If it’s not found there exists a gcc attribute which is not compatible with MSVC. It’s fixed upstream but no release was scheduled I will prepare a patch which solves this. Until then this PR must not be merged.

**Comment by tobiaskohlbau** [comment]
I've fixed the stated issue with `bzip2` and checked that this bump builds for `arm-uwp`, `x64-uwp` as well.
 Should be ready to merge now.

**Review by janisozaur — APPROVED** [review]
Seems to compile correctly and all the issues I raised were addressed.

**Comment by ras0219-msft** [comment]
Thanks for the PR and thanks for reviewing it!

**Pull request merged**